File: DwindleDating/Backend/Models/Player.m
//
//  Player.m
//  DwindleDating
//
//  Created by Yunas Qazi on 4/16/15.
//  Copyright (c) 2015 infinione. All rights reserved.
//

#import "Player.h"

@implementation Player

- (instancetype)initWithDict:(NSDictionary*)dict{
    self = [self initWithFBId:dict[@"fb_id"] andImagePath:dict[@"pic_path"]];
    return self;
}

- (instancetype)initWithFBId:(NSString*)fbId andImagePath:(NSString*)imgPath{
    self = [super init];
    if (self) {
        self.fbId = [NSString stringWithString:fbId];
        self.imgPath = [NSURL URLWithString:imgPath];
        self.galleryImages = [NSMutableArray new];
        [self.galleryImages addObject:self.imgPath];
    }
    return self;
}

- (instancetype)init{
    self = [super init];
    if (self) {
        
    }
    return self;
}


- (void) addImageUrlToGallery:(NSString*)imgPath{
     [self.galleryImages addObject:[NSURL URLWithString:imgPath]];
}

@end

File: DwindleDating/Backend/Models/Player.swift
//
//  Player.swift
//  DwindleDating
//
//  Created by Yunas Qazi on 4/16/15.
//  Copyright (c) 2015 infinione. All rights reserved.
//

import Foundation



class Player: NSObject {
    
    var fb_id: NSString // no need (!). It will be initialised from controller
    var imgPath: NSURL
    
    init(fromFbId fbId: NSString, fromImagePath imgPath: NSString) {
        self.fb_id = fbId
        self.imgPath = NSURL(string: imgPath)!
        super.init()
    }
    
    convenience override init() {
        self.init(fromFbId:"", fromImagePath:"") // calls above mentioned controller with default name
    }
}